### ISSUE TITLE
v1.8.0: missing RBAC to GET resource servicemonitors

### ISSUE THREAD

**Issue by wjentner** [rook/rook#9398]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

Upgrade process fails with error:
```
2021-12-11 23:27:14.933179 I | op-k8sutil: Reporting Event rook-ceph:rook-ceph Warning:ReconcileFailed:failed to reconcile cluster "rook-ceph": failed to configure local ceph cluster: failed to create cluster: failed to start ceph mgr: failed to enable mgr services: failed to enable service monitor: service monitor could not be enabled: failed to retrieve servicemonitor. servicemonitors.monitoring.coreos.com "rook-ceph-mgr" is forbidden: User "system:serviceaccount:rook-ceph:rook-ceph-system" cannot get resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "rook-ceph"
```

**Expected behavior:**

Successfully reconciling the cluster.

**How to reproduce it (minimal and precise):**

Helm chart version 1.8.0 with default values.yaml

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): Debian GNU/Linux 10 (buster)
* Kernel (e.g. `uname -a`): Linux kube-worker4 4.19.0-18-amd64 #1 SMP Debian 4.19.208-1 (2021-09-29) x86_64 GNU/Linux
* Cloud provider or hardware configuration: bare-metal
* Rook version (use `rook version` inside of a Rook Pod): 1.8.0
* Storage backend version (e.g. for ceph do `ceph -v`): 16.2.6
* Kubernetes version (use `kubectl version`): 1.21.7
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): kubeadm, bare-metal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)): healthy.

**Comment by ibotty** [comment]
Re b85df0f: It would be great to ship the roles that allow the servicemonitors instead of disabling them.

**Comment by blackliner** [comment]
TL&DR: if you have `monitoring.enabled = true` in your `rook-ceph-cluster.yaml' , do the same for the operator, aka `rook-ceph.yaml`

**Comment by parth-gr** [comment]
> TL&DR: if you have `monitoring.enabled = true` in your `rook-ceph-cluster.yaml` , do the same for the operator, aka `rook-ceph.yaml`

What do you mean by rook-cceph.yaml @blackliner

**Comment by stemid** [comment]
> TL&DR: if you have `monitoring.enabled = true` in your `rook-ceph-cluster.yaml` , do the same for the operator, aka `rook-ceph.yaml`

I'm also having this permissions issue in 1.8 but I don't understand your answer. I understand that rook-ceph-cluster.yaml is the CephCluster spec, and it does have spec.monitoring.enabled=true. But I'm using the rook/deploy/examples/operator.yaml and if I add spec.monitoring I get a ValidationError saying monitoring is not valid.

**Comment by bumarcell** [comment]
What about clusterrole `rook-ceph-cluster-mgmt` which is deployed by the operator and doesn't have a template for adding resources?
